test/unit/attributes.js

```diff
@@ -1107,6 +1107,71 @@ QUnit.test( "val(select) after form.reset() (Bug #2551)", function( assert ) {
 	jQuery( "#kk" ).remove();
 } );
 
+QUnit.test( "select.val(space characters) (gh-2978)", function( assert ) {
```

Comment:
Yeah, ajax looks better, i have shown that [before](https://github.com/jquery/jquery/pull/2810#discussion_r49638025), right now though, i'm low on motivation about this, since most of the team don't see as importance in it, which is a shame.